ConsoleApp1/Lab1/Program.cs

```diff
@@ -1,40 +1,163 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Dynamic;
+using System.Net.Sockets;
+using System.Xml;
+
 namespace ConsoleApp1.Lab1
 {
-    internal class Lab1
+    internal class Program
     {
         static void Main(string[] args)
         {
-            var p1 = new CreateUser("jopka");
-            var p2 = new CreateUser("jorgik");
-            Console.WriteLine(p1.UserName + " " + p1.GamesCount + " " + p1.CurrentRating);
-            Console.WriteLine(p2.UserName + " " + p2.GamesCount + " " + p2.CurrentRating);
+            var playerOne = new GameAccount("Bender");
+            var playerTwo = new GameAccount("STEPAN");
+            var gameone = new Game(playerOne, playerTwo, 10); 
+            gameone.PowerCheck();
+            gameone.PowerCheck();
+            gameone.PowerCheck();
+            gameone.PowerCheck();
+            gameone.PowerCheck();
+            Console.WriteLine(playerOne.GetStats());
+            Console.WriteLine(playerTwo.GetStats());
         }
 
-         class CreateUser
+        public class GameAccount
         {
-            public string UserName {get;}
-            public int  GamesCount {get;}
-            public  int CurrentRating {get;}
+            public string UserName { get; }
+            public int GamesCount { get; }
+
+            public int GameIndex
+            {
+                get
+                {
+                    int gameId = 1;
+                    foreach (var t in allOperations)
+                    {
+                        gameId += t.GameIndex;
+                    }
+            
+                    return gameId;
+                }
+                
+            }
+
+            public int CurrentRating
+            {
+                get
+                {
+                    int currentRating = 1000;
+                    foreach (var t in allOperations)
+                    {
+                        currentRating = t.Rating;
```

Comment:
You assign the current rating the value of the rating you are playing for. And you need to do manipulations with the rating you are playing for. 
**currentRating = t.Rating;** != **currentRating += t.Rating;**

Reply:
Thanks a lot, Sir Rezenkov